exporters/jaeger-thrift/src/test/java/io/opentelemetry/exporter/jaeger/thrift/JaegerThriftIntegrationTest.java

```diff
@@ -35,37 +37,48 @@ class JaegerThriftIntegrationTest {
 
   private static final int QUERY_PORT = 16686;
   private static final int THRIFT_HTTP_PORT = 14268;
+
+  private static final int THRIFT_UDP_PORT = 6831;
   private static final int HEALTH_PORT = 14269;
   private static final String SERVICE_NAME = "E2E-test";
   private static final String JAEGER_URL = "http://localhost";
 
   @Container
   public static GenericContainer<?> jaegerContainer =
       new GenericContainer<>("ghcr.io/open-telemetry/opentelemetry-java/jaeger:1.32")
-          .withExposedPorts(THRIFT_HTTP_PORT, QUERY_PORT, HEALTH_PORT)
+          .withExposedPorts(THRIFT_HTTP_PORT, THRIFT_UDP_PORT, QUERY_PORT, HEALTH_PORT)
           .withLogConsumer(new Slf4jLogConsumer(LoggerFactory.getLogger("jaeger")))
           .waitingFor(Wait.forHttp("/").forPort(HEALTH_PORT));
 
-  @Test
-  void testJaegerIntegration() {
-    OpenTelemetry openTelemetry = initOpenTelemetry();
+  @ParameterizedTest
+  @ValueSource(booleans = {false, true})
+  void testJaegerIntegration(boolean udp) {
+    OpenTelemetry openTelemetry = initOpenTelemetry(udp);
     imitateWork(openTelemetry);
     Awaitility.await()
         .atMost(Duration.ofSeconds(30))
         .until(JaegerThriftIntegrationTest::assertJaegerHasATrace);
   }
 
-  private static OpenTelemetry initOpenTelemetry() {
-    Integer mappedPort = jaegerContainer.getMappedPort(THRIFT_HTTP_PORT);
+  private static OpenTelemetry initOpenTelemetry(boolean udp) {
+    JaegerThriftSpanExporterBuilder jaegerExporter = JaegerThriftSpanExporter.builder();
+
+    if (udp) {
+      int mappedPort = jaegerContainer.getMappedPort(THRIFT_UDP_PORT);
+      try {
+        jaegerExporter.setThriftSender(new UdpSender("localhost", mappedPort, 0));
+      } catch (TTransportException e) {
+        throw new IllegalStateException(e);
```

Comment:
I wonder if this can have any negative impact on the test stability if the server is not ready at this stage. IIRC jaeger-client was resilient for this scenario.

Reply:
We're just using jaeger-client under the hood so the same reliance would apply. But we're also blocking on Jaeger's health endpoint when starting the test, so also hope that it is reflecting the server is ready :P